### DISCUSSION TITLE
Changing the properties of the tiff file

### DISCUSSION THREAD

**Discussion by SikangSHU** [libvips/libvips#2587] [Q&A]
Dear Professor:

After I use the 'vipsheader' to see the properties of the tiff file, the query results for the two files are as follows.

```
$ vipsheader E:\\gray1_1D.tif 
E:\\gray1_1D.tif: 66607x53298 uchar, 1 band, b-w, tiffload`
$ vipsheader E:\\1.tif 
E:\\1.tif: 966x711 ushort, 1 band, grey16, tiffload`
```

I'd like to set the properties of the first file('gray1_1D.tif') to be the same as the ones of the second file('1.tif'), which means change 'b-w' to 'grey16' so that the two tiff file have the same properties. 

Can you help me? How can I do that? Thank you very much!

**Comment by jcupitt** [comment]
Hi @SikangSHU,

No need to call me professor. You can use `colourspace` for this, eg.:

```
vips colourspace gray1_1D.tif new.tif grey16
```

**Reply by SikangSHU** [reply]
Thank you! 
Maybe I have a misunderstanding about metadata. I think 'vipsheader' can only read the properties of a tiff file. So I'd like to ask how can I read the tiff file metadata. Can I realize it by pyvips?

**Reply by jcupitt** [reply]
libvips doesn't try to support all TIFF metadata. You can see what is supported with `vipsheader -a`, for example:

```
$ vipsheader -a mri.tif 
mri.tif: 256x256 uchar, 1 band, b-w, tiffload
width: 256
height: 256
bands: 1
format: uchar
coding: none
interpretation: b-w
xoffset: 0
yoffset: 0
xres: 47.2441
yres: 47.2441
filename: mri.tif
vips-loader: tiffload
n-pages: 1
image-description: MatLab PGMWRITE file, saved 27-Aug-96
resolution-unit: in
orientation: 1
```

**Comment by SikangSHU** [comment]
Thank you for your patient answering! I had tried 'vipsheader -a' but found that it didn't have the color information.
I need to know color information(or stain information) because  I'd like to convert the first image(Figure 1) which is a tiff file to the flurescence image so that the effect can be the same as that in Figure 2 when I open it with Qupath. I guess that I should view and modify the metadata which includes colour information(or stain information) of the two tif files(two images), so I need to know the metadata including color information(or stain information). Do you know how to do that? Thank you!
![Figure1](https://user-images.githubusercontent.com/95743639/146776410-0a2e5a27-59c9-4df6-92c7-4fa31dad5697.png)
![Figure2](https://user-images.githubusercontent.com/95743639/146776481-f5c59158-f3ff-48d4-859a-9344be1fb2bf.jpeg)
